src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -375,6 +376,39 @@ public Repository getOrCheckoutRepository(VcsRepositoryUrl repoUrl, String targe
         return getOrCheckoutRepository(repoUrl, localPath, pullOnGet);
     }
 
+    /**
+     * Get the local repository for a given remote repository URL. If the local repo does not exist yet, it will be checked out.
+     * After retrieving the repository, the commit for the given hash will be checked out.
+     *
+     * @param repository the repository to check out the commit in
+     * @param commitHash the hash of the commit to check out
+     * @return the repository checked out at the given commit
+     */
+    public Repository checkoutRepositoryAtCommit(Repository repository, String commitHash) {
+        try (Git git = new Git(repository)) {
+            git.checkout().setName(commitHash).call();
+        }
+        catch (GitAPIException e) {
+            throw new GitException("Could not checkout commit " + commitHash + " in repository located at  " + repository.getLocalPath(), e);
+        }
+        return repository;
+    }
+
+    /**
+     * Get the local repository for a given remote repository URL. If the local repo does not exist yet, it will be checked out.
```

Comment:
```suggestion
     * Get the local repository for a given remote repository URL.
     * <p>
     * If the local repo does not exist yet, it will be checked out.
```